velox/functions/prestosql/aggregates/GeometricMeanAggregate.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cmath>
+#include "velox/exec/SimpleAggregateAdapter.h"
+#include "velox/functions/prestosql/aggregates/AggregateNames.h"
+
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::aggregate::prestosql {
+
+namespace {
+
+template <typename T>
+class GeometricMeanAggregate {
+ public:
+  using InputType = Row<T>;
+
+  using IntermediateType =
+      Row</*logSum*/ double,
+          /*count*/ int64_t>;
+
+  using OutputType = double;
+
+  static bool toIntermediate(
+      exec::out_type<Row<double, int64_t>>& out,
+      exec::arg_type<T> in) {
+    out.copy_from(std::make_tuple(std::log(in), 1));
+    return true;
+  }
+
+  struct AccumulatorType {
+    // Sum of `log(value)`.
+    double logSum_{0};
+    // Count of values.
+    int64_t count_{0};
+
+    AccumulatorType() = delete;
+
+    explicit AccumulatorType(HashStringAllocator* /*allocator*/) {}
+
+    void addInput(HashStringAllocator* /*allocator*/, exec::arg_type<T> data) {
+      logSum_ += std::log(data);
+      count_ = checkedPlus<int64_t>(count_, 1);
+    }
+
+    void combine(
+        HashStringAllocator* /*allocator*/,
+        exec::arg_type<Row<double, int64_t>> other) {
+      // Use VELOX_USER_CHECK here to make aggregation fuzzer happy.
```

Comment:
Remove this comment and change registration code to not register companion functions.

Reply:
done